packages/adapter-components/src/elements/swagger/type_elements/swagger_parser.ts

```diff
@@ -154,17 +154,33 @@ export const toSchema = (
   swaggerVersion === SwaggerVersion.V2 ? toSchemaV2(def as V2Def) : toSchemaV3(def as V3Def)
 )
 
-export const getParsedDefs = async (swaggerPath: string, loadedSwagger?: LoadedSwagger):
-  Promise<SchemasAndRefs> => {
+export const getParsedDefs = async ({
+  swaggerPath,
+  loadedSwagger,
+  additionalTypes,
+}:{
+  swaggerPath: string
+  loadedSwagger?: LoadedSwagger
+  additionalTypes?: string[]
+}):Promise<SchemasAndRefs> => {
   const swagger = loadedSwagger ?? await loadSwagger(swaggerPath)
 
   const swaggerVersion = getSwaggerVersion(swagger)
   const responseSchemas = _.pickBy(
     _.mapValues(swagger.document.paths, def => toSchema(swaggerVersion, def.get?.responses?.[200])),
     isDefined,
   )
+  const additionalSchemas = isV3(swagger.document)
+    ? _.pickBy(
+      swagger.document.components?.schemas,
+      (_value, key) => additionalTypes?.includes(key)
+    )
+    : _.pickBy(
+      swagger.document.definitions,
+      (_value, key) => additionalTypes?.includes(key)
+    )
```

Comment:
nits:
1. can we convert `additionalTypes` to a set this is more efficient if there are many?
2. maybe this can be shortened to
```suggestion
  const additionalSchemas = _.pickBy(
    isV3(swagger.document)
      ? swagger.document.components?.schemas
      : swagger.document.definitions,
    (_value, key) => additionalTypes?.includes(key)
  )
```